### ISSUE TITLE
_find_calling_layer relies on naming conventions

### ISSUE THREAD

**Issue by stub42** [juju-solutions/layer-status#5]
It seems that _find_calling_layer relies on charms and layers using specific naming conventions and placing files in specific locations. This seems fragile. An alternative would be to match the calling filename to a layer or interface using the .build-manifest file.

**Comment by johnsca** [comment]
Agreed that the heuristic is faulty, but unfortunately, the `.build-manifest` is not included in published charms (but see juju/charm-tools#404).  +1 to using the manifest once that issue is addressed and it's available.

On the plus side, the layer name is only used to try to do a better job prioritizing status messages when there's a conflict, so it's not the end of the world if it fails (it falls back to LIFO).

**Comment by stub42** [comment]
Yeah, I was thinking later that blocked is blocked, and the heuristic for determining which blocked message is more important isn't important and unlikely to actually be an issue.

**Comment by stub42** [comment]
(jumped the gun, since the referenced issue means we may want to change this)
